### PR TITLE
Add a data persistence layer

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/core/pull/1460#discussion_r146764601
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Bridge/Doctrine/Common/DataPersister.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($data): bool
+    {
+        return (bool) $this->getManager($data);
```
Comment (dunglas) [inline src/Bridge/Doctrine/Common/DataPersister.php]: Wouldn't `null !== $this->getManager($data)` be less weird?

File: src/Bridge/Doctrine/EventListener/WriteListener.php [hunk 2]
```diff
@@ -1,87 +0,0 @@
-<?php
```
Comment (dunglas) [inline src/Bridge/Doctrine/EventListener/WriteListener.php]: We cannot drop it right now (BC promise), this class must be deprecated instead.

File: src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataPersisterPass.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers data persisters.
+ *
+ * @internal
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataPersisterPass.php]: We'll be able to drop this class when we'll upgrade to DI 3.4: http://symfony.com/blog/new-in-symfony-3-4-simpler-injection-of-tagged-services

File: src/Bridge/Doctrine/EventListener/WriteListener.php (hunk 2 above)
Comment (meyerbaptiste) [reply src/Bridge/Doctrine/EventListener/WriteListener.php]: So should we also keep the definition of the tagged service and depreciate it?

File: src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataPersisterPass.php (hunk 3 above)
Comment (meyerbaptiste) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataPersisterPass.php]: Yep, I saw that. Do you suggest to delete it?

File: src/Bridge/Doctrine/EventListener/WriteListener.php (hunk 2 above)
Comment (dunglas) [reply src/Bridge/Doctrine/EventListener/WriteListener.php]: I'm not sure about that because it's a public service so it will always be executed.

**Comment by meyerbaptiste** [comment]
Done @dunglas

**Comment by dunglas** [comment]
Thanks @meyerbaptiste

**Comment by dunglas** [comment]
Don't forget the docs PR :D

**Pull request merged**